src/main/scala/dpla/ingestion3/mappers/providers/NaraExtractor.scala

```diff
@@ -0,0 +1,256 @@
+    } yield name
+
+    organizationalContributors ++ personalContributors
+  }
+
+  def creator(xml: NodeSeq): Seq[String] = {
+
+    //TODO: not handling multiple display values. haven't found example yet.
+    val organizationalCreators = for (
+      name <- xml \\ "creatingOrganizationArray" \ "creatingOrganization" \ "creator" \ "termName"
+    ) yield name.text
+
+    val individualCreators = for (
+      name <- xml \\ "creatingIndividualArray" \ "creatingIndividual" \ "creator" \ "termName"
+    ) yield name.text
+
+    if (organizationalCreators.nonEmpty) organizationalCreators else individualCreators
+  }
+
+
+  def nodeToDateString(nodeOption: Option[Node]): Option[String] = nodeOption match {
+    case None => None
+    case Some(node) =>
+      val year = (node \ "year").text
+      val month = (node \ "month").text
+      val day = (node \ "day").text
+
+      (year, month, day) match {
+        case (y, m, d) if y.isEmpty => None
+        case (y, m, d) if m.isEmpty => Some(year)
+        case (y, m, d) if d.isEmpty => Some(s"$year-$month")
+        case (y, m, d) => Some(s"$year-$month-$day")
+      }
+  }
+
+  def getDisplayDate(start: Option[Node], end: Option[Node]): Option[String] = {
+    if (start.isEmpty && end.isEmpty) {
+      None
+    } else {
+      val startString = start.map(_.text).getOrElse("unknown")
+      val endString = end.map(_.text).getOrElse("unknown")
+      Some(s"$startString/$endString")
+    }
+  }
+
+  def simpleDate(nodeSeq: NodeSeq): Seq[EdmTimeSpan] =
+    nodeSeq.map(node => EdmTimeSpan(originalSourceDate = nodeToDateString(Some(node))))
+
+  def date(xml: NodeSeq): Seq[EdmTimeSpan] = {
+
+    val coverageDates = for {
+      coverageDate <- xml \\ "coverageDates"
+      coverageStartDate = (coverageDate \ "coverageStartDate").headOption
+      coverageEndDate = (coverageDate \ "coverageEndDate").headOption
+      //dateQualifier = coverageDate \ "dateQualifier" //todo not sure what to do with qualifier in EDTF
+      if coverageStartDate.nonEmpty || coverageEndDate.nonEmpty
+
+    } yield {
+      val displayDate = getDisplayDate(coverageStartDate, coverageEndDate)
+      EdmTimeSpan(
+        originalSourceDate = displayDate, //this is cheating?
+        begin = nodeToDateString(coverageStartDate),
+        end = nodeToDateString(coverageEndDate),
+        prefLabel = displayDate
+      )
+    }
+
+    val copyrightDates = simpleDate(xml \\ "copyrightDateArray" \ "proposableQualifiableDate")
+    val productionDates = simpleDate(xml \\ "productionDateArray" \ "proposableQualifiableDate")
+
+    //todo not sure what to do with logicalDate
+    val broadcastDates = simpleDate(xml \\ "broadcastDateArray" \ "proposableQualifiableDate")
+    val releaseDates = simpleDate(xml \\ "releaseDateArray" \ "proposableQualifiableDate")
+
+    Seq(
+      coverageDates,
+      copyrightDates,
+      productionDates,
+      broadcastDates,
+      releaseDates
+    )
+      .collectFirst({ case x => x })
```

Comment:
It gets the first result that's not None. So, the first execution that returns a value. There's a better way to do this, though, so I'll check that in instead.